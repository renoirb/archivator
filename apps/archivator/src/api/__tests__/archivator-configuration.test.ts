import * as path from 'path'

import { ArchivatorConfiguration } from '..'

describe('ArchivatorConfiguration', () => {
  const configFile: string = path.resolve(
    __dirname,
    'sample-repo',
    'archivator.json',
  )

  let _oldEnv: typeof process.env

  beforeEach(() => {
    _oldEnv = process.env
    process.env = {}
  })

  afterEach(() => {
    process.env = _oldEnv
  })

  it('can load configuration using loadFromConfigurationFile', (done: jest.DoneCallback) => {
    expect(() => {
      ArchivatorConfiguration.loadFromConfigurationFile(configFile)
    }).not.toThrow()

    done()
  })

  it('is possible to get enumeration of archives', (done: jest.DoneCallback) => {
    const subject = ArchivatorConfiguration.loadFromConfigurationFile(
      configFile,
    )

    expect(subject).toHaveProperty('archives')
    expect(subject.archives).toMatchSnapshot()
    expect(subject).toHaveProperty('archivesByName')
    expect(subject.archivesByName).toMatchSnapshot()

    done()
  })

  it('is possible to get one archive by asking getArchiveByName', (done: jest.DoneCallback) => {
    const subject = ArchivatorConfiguration.loadFromConfigurationFile(
      configFile,
    )

    const archiveData = subject.getArchiveByName('paleo-recipes')
    expect(archiveData).toMatchSnapshot()

    done()
  })

  it('is possible to get archive CSV lines by asking createIterable', (done: jest.DoneCallback) => {
    const subject = ArchivatorConfiguration.loadFromConfigurationFile(
      configFile,
    )

    const archiveIndexLines = subject.createIterable('research-papers')
    const collection = [...archiveIndexLines]
    expect(collection).toMatchSnapshot()

    done()
  })
})
