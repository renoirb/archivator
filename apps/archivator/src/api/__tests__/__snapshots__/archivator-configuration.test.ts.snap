// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArchivatorConfiguration is possible to get archive CSV lines by asking createIterable 1`] = `
Array [
  Object {
    "archive": null,
    "selector": "body",
    "truncate": "script,style,noscript,template",
    "url": "https://ponyfoo.com/articles/const-variables-not-immutable",
  },
  Object {
    "archive": null,
    "selector": "body",
    "truncate": "script,style,noscript,template",
    "url": "http://omergil.blogspot.com.br/2017/02/web-cache-deception-attack.html",
  },
  Object {
    "archive": null,
    "selector": "body",
    "truncate": "script,style,noscript,template",
    "url": "https://alistapart.com/article/application-cache-is-a-douchebag/index.html",
  },
]
`;

exports[`ArchivatorConfiguration is possible to get enumeration of archives 1`] = `
Array [
  Object {
    "archiveFolder": "paleo-recipes",
    "archiveName": "paleo-recipes",
  },
  Object {
    "archiveFolder": "some/nested/folder/research-papers",
    "archiveName": "research-papers",
  },
]
`;

exports[`ArchivatorConfiguration is possible to get enumeration of archives 2`] = `
Map {
  "paleo-recipes" => Object {
    "archiveFolder": "paleo-recipes",
    "archiveName": "paleo-recipes",
  },
  "research-papers" => Object {
    "archiveFolder": "some/nested/folder/research-papers",
    "archiveName": "research-papers",
  },
}
`;

exports[`ArchivatorConfiguration is possible to get one archive by asking getArchiveByName 1`] = `
Object {
  "archiveFolder": "paleo-recipes",
  "archiveName": "paleo-recipes",
}
`;
